tests/bugs/gh-4556.slang

```diff
@@ -0,0 +1,21 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHECK):-dx12 -compute -output-using-type -shaderobj
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -compute -output-using-type -shaderobj
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=CHECK):-vk -glsl -compute -output-using-type -shaderobj
```

Comment:
I am getting a validation error for the 2nd and 3rd test.
The error message is following,
```
error: line 51: Expected float scalar or vector type as Result Type: FConvert
  %b = OpFConvert %mat3v4double %12
```
You may want to add the following test to catch the problem.
```
//TEST(compute):SIMPLE(filecheck=SPIRV): -entry computeMain -target spirv-asm -stage compute
//SPIRV-NOT: Validation of generated SPIR-V failed
```
```

Reply:
I am trying to figure out why CI is fine and my machine is not fine.
I have a higher version of Vulkan-SDK than what CI uses, but I am not sure if that's the real cause or not yet.